pytest/tests/sanity/validator_switch_key_quick.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+# Starts three validating nodes and one non-validating node
+# Set a new validator key that has the same account id as one of
+# the validating nodes. Stake that account with the new key
+# and make sure that the network doesn't stall even after
+# the non-validating node becomes a validator.
+
+import unittest
+import sys, time
+import pathlib
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+
+from configured_logger import logger
+from cluster import start_cluster
+
+EPOCH_LENGTH = 20
+TIMEOUT = 100
+
+
+class ValidatorSwitchKeyQuickTest(unittest.TestCase):
+
+    def test_validator_switch_key_quick(self):
+        # It is important for the non-validating node to already track shards
+        # that it will be assigned to when becoming a validator.
+        config_map = {
+            2: {
+                "tracked_shards": [0, 2],
```

Comment:
This does not do what you think it does. `tracked_shards` is a total mess. It acts a as binary flag. If it's empty it means track no shards or track tracked shards for validators. If it's non-empty it means track all shards. This is why typically you see it set to [0] in test. Welcome to nearcore ;)